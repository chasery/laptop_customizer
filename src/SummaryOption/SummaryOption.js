import React from "react";
import { USCurrencyFormat } from "../USCurrencyFormat";
import "./SummaryOption.css";

class SummaryOption extends React.Component {
  render() {
    const { feature, selectedOptionName, selectedOptionCost } = this.props;
    return (
      <div className="summary__option">
        <div className="summary__option__label">{feature}</div>
        <div className="summary__option__value">{selectedOptionName}</div>
        <div className="summary__option__cost">
          {USCurrencyFormat.format(selectedOptionCost)}
        </div>
      </div>
    );
  }
}

export default SummaryOption;
