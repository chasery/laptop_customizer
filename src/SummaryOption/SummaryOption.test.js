import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SummaryOption from "./SummaryOption";

describe("SummaryOption component", () => {
  it("renders without errors", () => {
    const wrapper = shallow(<SummaryOption />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
