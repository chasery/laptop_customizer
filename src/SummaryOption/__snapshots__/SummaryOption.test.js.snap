// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryOption component renders without errors 1`] = `
<div
  className="summary__option"
>
  <div
    className="summary__option__label"
  />
  <div
    className="summary__option__value"
  />
  <div
    className="summary__option__cost"
  >
    $NaN
  </div>
</div>
`;
